block_server.py

```diff
@@ -367,12 +365,17 @@ def write(self, reason, value):
                 with self.write_lock:
                     self.write_queue.append((self.save_inactive_config, (data, True), "SAVING_NEW_COMP"))
             elif reason == BlockserverPVNames.DELETE_CONFIGS:
-                self._config_list.delete_configs(convert_from_json(data))
+                with self.write_lock:
+                    self.write_queue.append((self._config_list.delete_configs, (convert_from_json(data),),
+                                             "DELETE_CONFIGS"))
             elif reason == BlockserverPVNames.DELETE_COMPONENTS:
-                self._config_list.delete_configs(convert_from_json(data), True)
+                with self.write_lock:
+                    self.write_queue.append((self._config_list.delete_configs, (convert_from_json(data), True),
+                                             "DELETE_COMPONENTS"))
             elif reason == BlockserverPVNames.BUMPSTRIP_AVAILABLE_SP:
                 self.bumpstrip = data
-                self.update_bumpstrip_availability()
+                with self.write_lock:
+                    self.write_queue.append((self.update_bumpstrip_availability(), None, "UPDATE_BUMPSTRIP"))
```

Comment:
This call should pass in `self.update_bumpstrip_availability`. Because of the brackets, I think the function will be executed immediately and the return value passed to the queue. Correct?